actors.md

```diff
@@ -4,6 +4,42 @@ Any implementations of the Filecoin actors must be exactly byte for byte compati
 
 This spec decsribes a set of actors that operate within the [Filecoin State Machine](state-machine.md). All types are defined in [the basic type encoding spec](data-structures.md#basic-type-encodings).
 
+## Init Actor
+The init actor is responsible for creating new actors on the filecoin network. This is a built-in actor and cannot be replicated. In the future, this actor will be responsible for loading new code into the system (for user programmable actors).
+
+```go
+type InitActor struct {
+    NextActorID uint64
+}
+```
+
+#### Exec
+
+Parameters:
+
+- code Cid
+- params []byte
+
+TODO: the type of params should probably just be variadic set of arguments. Making it be a `[]byte` here kinda sucks
+
+```go
+func Exec(code Cid, params []byte) Address {
+    actorID := self.NextActorID
+    self.NextActorID++
+    
+    if code == InitActorCodeCid {
+        Fatal("cannot launch additional init actor instance")
+    }
+    
+    addr := VM.CreateActor(code, msg.Value, DecodeParams(params))
+    
+    // TODO: where is this mapping stored?
+    SetIDMapping(actorID, addr)
+    
+    return addr
```

Comment:
how do I get the ID from the addr?

Reply:
I was thinking that nodes would just create and store a reverse index locally. Storing a two-way mapping on-chain seems expensive.

Reply:
alternatively, I was talking with @frrist about a setup where the primary state tree is a `map[ID]Actor`, and separately we store a `map[Address]ID`. This would allow users to shift around keys for their accounts, and maybe other things.

Reply:
I think we said a `map[PubKey]ID` when we initially talked about this, is that what we mean by `Address` here?

Reply:
I think using the `ID` as primary identifier in the state tree would be neat. Having a `map[Address]ID` around, should also address any reordering concerns that could appear due to different `ID` assignments. We could store this mapping in the `InitActor`, given it is the actor that actually adds entries to this mapping. 

Allowing key changes would need some more thought, because if I simply change my key there is no information about my new address (the one derived from my new key) in the `Addresss -> ID` map. So this would require having at least an `AddMapping(from Address, to ID)` call to the `InitActor` (or the place that manages this mapping)

Reply:
Yeah, i'm in support of IDs being used as the primary mapping.

Reply:
done